File: androidApp/src/main/java/press/home/HomeModule.kt
package press.home

import dagger.Module
import dagger.Provides
import me.saket.press.shared.home.HomePresenter
import me.saket.press.shared.home.SharedHomeComponent

@Module
object HomeModule {
  @Provides
  fun presenter(): HomePresenter.Factory {
    return HomePresenter.Factory { SharedHomeComponent.presenter(it) }
  }
}

File: native/mac/mac/Home/HomeModule.swift
//
// Created by Saket Narayan on 4/26/20.
// Copyright (c) 2020 Saket Narayan. All rights reserved.
//

import Foundation
import Swinject
import shared

class HomeComponent: Assembly {

  func assemble(container: Container) {
    container.register(HomeView.self) { r in
      HomeView(presenterFactory: r.resolve(HomePresenterFactory.self)!)
    }

    container.register(HomePresenterFactory.self) { r in
      class Factory: HomePresenterFactory {
        func create(args: HomePresenter.Args) -> HomePresenter {
          SharedHomeComponent.Companion().presenter(args: args)
        }
      }
      return Factory()
    }
  }
}
